iree/vm/native_module_packing.h

```diff
@@ -413,14 +431,40 @@ struct ParamUnpack<const ref<T>> {
 // An `iree.byte_buffer` containing a string.
 // The string view is aliased directly into the underlying byte buffer.
 template <>
-struct ParamUnpack<absl::string_view> {
-  using storage_type = absl::string_view;
+struct ParamUnpack<iree_string_view_t> {
+  using storage_type = iree_string_view_t;
+  static void Load(Status& status, params_ptr_t& ptr, storage_type& out_param) {
+    auto* reg_ptr = reinterpret_cast<iree_vm_ref_t*>(ptr);
+    ptr += sizeof(iree_vm_ref_t);
+    if (reg_ptr->type == ref_type_descriptor<iree_vm_buffer_t>::get()->type) {
+      auto byte_span = reinterpret_cast<iree_vm_buffer_t*>(reg_ptr->ptr)->data;
+      out_param = iree_make_string_view(
+          reinterpret_cast<const char*>(byte_span.data), byte_span.data_length);
+    } else if (IREE_UNLIKELY(reg_ptr->type != IREE_VM_REF_TYPE_NULL)) {
+      status = iree_make_status(
+          IREE_STATUS_INVALID_ARGUMENT,
+          "parameter contains a reference to the wrong type; "
+          "have %.*s but expected %.*s",
+          (int)iree_vm_ref_type_name(reg_ptr->type).size,
+          iree_vm_ref_type_name(reg_ptr->type).data,
+          (int)ref_type_descriptor<iree_vm_buffer_t>::get()->type_name.size,
+          ref_type_descriptor<iree_vm_buffer_t>::get()->type_name.data);
+    } else {
+      // NOTE: empty string is allowed here!
+      out_param = iree_string_view_empty();
+    }
+  }
+};
+#if defined(IREE_HAVE_STD_STRING_VIEW)
```

Comment:
I wish we had a backport of string_view like span, but ok like this for now.

Reply:
yeah :( I debated but since this is interface is only used by 3rd party custom modules I felt like letting the user choose if they wanted to require c++17 or shim it was probably ok to start. Always something we/someone wanting it can add!